### PR TITLE
Python2 support added and put_object fix 

### PR DESCRIPTION
**Pull request by rajesh-project-n**

### CODE REVIEW

File: bolt/bolt_router.py [hunk 1]
```diff
@@ -80,20 +90,22 @@ class BoltSession(URLLib3Session):
...
+        return super(BoltSession, self)._get_pool_manager_kwargs(**extra_kwargs)
 
 
     def send(self, request):
         request.headers['Host'] = self._bolt_hostname
         for key in request.headers.keys():
-            request.headers[key] = str(request.headers[key])
-        return super().send(request)
+            if key == "Expect":
+                continue
```
Comment (kmushegi) [inline bolt/bolt_router.py:106]: Is this change py2 specific?

File: bolt/bolt_router.py [hunk 2]
```diff
@@ -109,10 +121,13 @@ def roundTime(dt=None, dateDelta=datetime.timedelta(minutes=1)):
...
+def _get_datatime_delta():
+    return datetime.timedelta(minutes=10)
+
 class BoltSigV4Auth(SigV4Auth):
-    def __init__(self, *args, bolt_timestamp_pin_duration = datetime.timedelta(minutes=10), **kwargs):
-        super().__init__(*args, **kwargs)
-        self.__bolt_timestamp_pin_duration = bolt_timestamp_pin_duration
+    def __init__(self, *args, **kwargs):
+        super(BoltSigV4Auth, self).__init__(*args, **kwargs)
+        self.__bolt_timestamp_pin_duration = datetime.timedelta(minutes=10)
```
Comment (kmushegi) [inline bolt/bolt_router.py:130]: No longer need ability to spec this into class? Always using 10 minutes?

**Review by kmushegi — COMMENTED** [review]
2 small questions otherwise lg

File: bolt/bolt_router.py (hunk 1 above)
Comment (rajesh-project-n) [reply bolt/bolt_router.py:106]: This is to make the put_object api call work.
This header and the `request.headers[key] = str(request.headers[key])` line was the cause for put object failure

Comment (kmushegi) [reply bolt/bolt_router.py:106]: Ok sounds good, for traceability I'd suggest things like this should go into the PR description so make it easier for reviewer so it'd be like

```
* py2 compat changes
* fix put_object ...
```

File: bolt/bolt_router.py (hunk 2 above)
Comment (kmushegi) [reply bolt/bolt_router.py:130]: sg!

**Review by MadhuPolu — APPROVED** [review]

**Review by MadhuPolu — APPROVED** [review]
Looks good to me

**Pull request closed without merging**